taxonomies/allergens.txt

```diff
@@ -13,13 +14,13 @@ en:gluten, cereals containing gluten, barley, barley malt flour, malted barley,
iutaleesta, täysjyväohrahiutaleet, täysjyväohrahiutaleita, täysjyväohrahiutaleista, täysjyväkaurahiutale, täysjyväkaurahiutaletta, täysjyväkaurahiutaleesta, täysjyväkaurahiutaleet, täysjyväkaurahiutaleita, täysjyväkaurahiutaleista, täysjyväspelttihiutale, täysjyväspelttihiutaletta, täysjyväspelttihiutaleesta, täysjyväspelttihiutaleet, täysjyväspelttihiutaleita, täysjyväspelttihiutaleista, vilja, viljaa, viljasta, viljat, viljoja, viljoista, gluteenia sisältävä vilja, gluteenia sisältävää viljaa, gluteenia sisältävästä viljasta, gluteenia sisältävät viljat, gluteenia sisältäviä viljoja, gluteenia sisältävistä viljoista, vehnäproteiini, vehnäproteiinia, vehnäproteiinista, vehnäproteiinit, vehnäproteiineja, vehnäproteiineista, durumvehnäproteiini, durumvehnäproteiinia, durumvehnäproteiinista, durumvehnäproteiinit, durumvehnäproteiineja, durumvehnäproteiineista, ruisproteiini, ruisproteiinia, ruisproteiinista, ruisproteiinit, ruisproteiineja, ruisproteiineista, ohraproteiini, ohraproteiinia, ohraproteiinista, ohraproteiinit, ohraproteiineja, ohraproteiineista, kauraproteiini, kauraproteiinia, kauraproteiinista, kauraproteiinit, kauraproteiineja, kauraproteiineista, spelttiproteiini, spelttiproteiinia, spelttiproteiinista, spelttiproteiinit, spelttiproteiineja, spelttiproteiineista, vehnätärkkelys, vehnätärkkelystä, vehnätärkkelyksestä, vehnätärkkelykset, vehnätärkkelyksiä, vehnätärkkelyksistä, durumvehnätärkkelys, durumvehnätärkkelys, durumvehnätärkkelystä, durumvehnätärkkelyksestä, durumvehnätärkkelykset, durumvehnätärkkelyksiä, durumvehnätärkkelyksistä, ruistärkkelys, ruistärkkelystä, ruistärkkelyksestä, ruistärkkelykset, ruistärkkelyksiä, ruistärkkelyksistä, ohratärkkelys, ohratärkkelyksestä, ohratärkkelykset, ohratärkkelyksiä, ohratärkkelyksistä, kauratärkkelys, kauratärkkelystä, kauratärkkelyksestä, kauratärkkelykset, kauratärkkelyksiä, kauratärkkelyksistä, spelttitärkkelys, spelttitärkkelystä, spelttitärkkelyksestä, spelttitärkkelykset, spelttitärkkelyksiä, spelttitärkkelyksistä, vehnämallasjauho, vehnämallasjauhoa, vehnämallasjauhosta, vehnämallasjauhot, vehnämallasjauhoja, vehnämallasjauhoista, durumvehnämallasjauho, durumvehnämallasjauhoa, durumvehnämallasjauhosta, durumvehnämallasjauhot, durumvehnämallasjauhoja, durumvehnämallasjauhoista, ruismallasjauho, ruismallasjauhoa, ruismallasjauhosta, ruismallasjauhot, ruismallasjauhoja, ruismallasjauhoista, ohramallasjauho, ohramallasjauhoa, ohramallasjauhosta, ohramallasjauhot, ohramallasjauhoja, ohramallasjauhoista, kauramallasjauho, kauramallasjauhoa, kauramallasjauhosta, kauramallasjauhot, kauramallasjauhoja, kauramallasjauhoista, spelttimallasjauho, spelttimallasjauhoa, spelttimallasjauhosta, spelttimallasjauhot, spelttimallasjauhoja, spelttimallasjauhoista, vehnäkuitu, vehnäkuituja, vehnäkuiduista, vehnänjyvä, vehnänjyvää, vehnänjyvästä, vehnänjyvät, vehnänjyviä, vehnänjyvistä, kauranjyvä, kauranjyvää, kauranjyvästä, kauranjyvät, kauranjyviä, kauranjyvistä, muita viljoja, muita viljatuotteita
```

Comment:
this list is really scary. at some point we'll need to find a better way to deal with all the possible forms on words in Finnish.

Reply:
It is. Couple of things that would improve it somewhat:

1. Using an allergen property in the ingredient taxonomy. That way the long list would be split into individual entries.
2. Detecting allergens inside compound words like `täysjyvä_vehnä_jauho`. Tricky to do automatically, but would allow only listing the allergens instead of every compound word where it appears.